### PR TITLE
Add baota to rsetup

### PR DESCRIPTION
**Pull request by RadxaPanda**
This PR adds Baota panel to the common tasks section of rsetup

### CODE REVIEW

File: src/usr/lib/rsetup/tui/task/baota/baota.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# shellcheck shell=bash
+
+__task_baota_uninstall() {
+    if yesno "Are you sure to uninstall Baota?"
+    then
+        if uninstall_baota
+        then
+            msgbox "Uninstall Baota success."
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/tui/task/baota/baota.sh:8]:
```suggestion
            msgbox "Baota is successfully uninstalled."
```

File: src/usr/lib/rsetup/tui/task/baota/baota.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        else
+            msgbox "Uninstall Baota failure."
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/tui/task/baota/baota.sh:10]:
```suggestion
            msgbox "An failure occurred when trying to uninstall Baota."  "$RTUI_PALETTE_ERROR"
```
$RTUI_PALETTE_ERROR can be used to display a red error message box.

File: src/usr/lib/rsetup/tui/task/baota/baota.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    menu_init
+    if [ -f /www/server/panel/BT-Panel ]
+    then
+        menu_add __task_baota_uninstall   "Uninstall Baota"
+    else
+
+        menu_add __task_baota_install     "Install Baota"
+    fi
+
+
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/tui/task/baota/baota.sh:32]:
```suggestion
```

File: src/usr/lib/rsetup/tui/task/baota/baota.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    fi
+}
+
+__task_baota() {
+    menu_init
+    if [ -f /www/server/panel/BT-Panel ]
+    then
+        menu_add __task_baota_uninstall   "Uninstall Baota"
+    else
+
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/tui/task/baota/baota.sh:28]:
```suggestion
```
Unnecessary new line

File: src/usr/lib/rsetup/tui/task/task.sh [hunk 5]
```diff
@@ -3,10 +3,12 @@
 # shellcheck source=src/usr/lib/rsetup/tui/task/docker/docker.sh
 source "/usr/lib/rsetup/tui/task/docker/docker.sh"
 source "/usr/lib/rsetup/tui/task/ssh/ssh.sh"
+source "/usr/lib/rsetup/tui/task/baota/baota.sh"
 
 __task() {
     menu_init
     menu_add __task_docker          "Docker"
     menu_add __task_ssh             "SSH"
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/tui/task/task.sh:11]:
```suggestion
    if $DEBUG
    then
        menu_add __task_docker          "Docker"
        menu_add __task_ssh             "SSH"
    fi
```
Those 2 will need to be reworked. Hide them for now.

**Review by RadxaYuntian — CHANGES_REQUESTED** [review]
1. edit `src/usr/lib/rsetup/tui/main.sh` to show __task, probably need to change the title to "Applications"
2. Use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) in our code base. Commit title should be `feat: add support for Baota`

**Comment by RadxaYuntian** [comment]
Your code also failed CI, so you probably did not install `pre-commit`.

File: src/usr/lib/rsetup/cli/baota.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# shellcheck shell=bash
+
+install_baota() {
+    __parameter_count_check 0 "$@"
+    curl -s https://download.bt.cn/install/install-ubuntu_6.0.sh | sudo bash /dev/stdin ed8484bec
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/cli/baota.sh:5]:
```suggestion
    sudo bash < <(curl -s "https://download.bt.cn/install/install-ubuntu_6.0.sh")
```
Please don't pin this to a specific commit. You know we ain't gonna come back, update, and validate for new commits, so just install the latest version, and redirect user's problem to Baota itself.

File: src/usr/lib/rsetup/cli/baota.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+uninstall_baota() {
+    __parameter_count_check 0 "$@"
+    curl -s https://download.bt.cn/install/bt-uninstall.sh | sudo bash /dev/stdin
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/cli/baota.sh:10]:
```suggestion
    sudo bash < <(curl -s "https://download.bt.cn/install/bt-uninstall.sh")
```
Just to look a little bit nicer.

File: src/usr/lib/rsetup/tui/task/baota/baota.sh [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+__task_baota_install() {
+    if yesno "Are you sure you want to install Baota?"
+    then
+        install_baota
+    fi
+}
+
+__task_baota() {
+    menu_init
+    if [ -f /www/server/panel/BT-Panel ]
```
Comment (RadxaYuntian) [inline src/usr/lib/rsetup/tui/task/baota/baota.sh:24]:
```suggestion
    if [[ -f /www/server/panel/BT-Panel ]]
```
`[ ]` should [generally be avoided](https://unix.stackexchange.com/questions/67172/vs-which-one-to-use-in-bash-scripts).

**Review by RadxaYuntian — CHANGES_REQUESTED** [review]

**Review by RadxaYuntian — CHANGES_REQUESTED** [review]
You can click `Re-request review` to notify me. Save a few types.
image
You can also squash your commits now.

**Comment by RadxaYuntian** [comment]
Also conventional commit's tag(`chore`, `feat`, `fix`, etc) and the first world of the real commit title is starting in lower case.

**Comment by RadxaPanda** [comment]
Squashed and pushed

**Review by RadxaYuntian — APPROVED** [review]
Lower case "A" next time.

**Pull request merged**